cloud/services/storage/keypair.go

```diff
@@ -0,0 +1,98 @@
+package storage
+
+import (
+	"errors"
+	"fmt"
+
+	infrastructurev1beta1 "github.com/outscale-dev/cluster-api-provider-outscale.git/api/v1beta1"
+	tag "github.com/outscale-dev/cluster-api-provider-outscale.git/cloud/tag"
+	osc "github.com/outscale/osc-sdk-go/v2"
+)
+
+type OscKeyPairInterface interface {
+	CreateKeyPair(spec *infrastructurev1beta1.OscKeypair, keypairName string) (*osc.KeypairCreated, error)
+	DeleteKeyPair(keypairName string) error
+	ReadKeyPair(keypairName string, keypairFingerprints string) (*osc.Volume, error)
+}
+
+func (s *Service) CreateKeyPair(spec *infrastructurev1beta1.OscKeypair, keypairName string) (*osc.KeypairCreated, error) {
+	keyPairName := spec.Name
+	publicKey := spec.PublicKey
+
+	keyPairRequest := osc.CreateKeypairRequest{
+		KeypairName: keyPairName,
+		PublicKey:   &publicKey,
+	}
+
+	oscApiClient := s.scope.GetApi()
+	oscAuthClient := s.scope.GetAuth()
+
+	keyPairResponse, httpRes, err := oscApiClient.KeypairApi.CreateKeypair(oscAuthClient).CreateKeypairRequest(keyPairRequest).Execute()
+	if err != nil {
+		fmt.Printf("Error with http result %s", httpRes.Status)
+		return nil, err
+	}
+
+	var resp osc.ReadKeypairsResponse
+
+	if len(resp.GetKeypairs()) < 1 {
+		return nil, fmt.Errorf("Unable to find key pair, please provide a better query criteria ")
+	}
+	if len(resp.GetKeypairs()) > 1 {
+		return nil, fmt.Errorf("Found to many key pairs, please provide a better query criteria ")
+	}
+
+	resourceIds := []string{*keyPairResponse.Keypair.KeypairName}
+	err = tag.AddTag("Name", keyPairName, resourceIds, oscApiClient, oscAuthClient)
+	if err != nil {
+		fmt.Printf("Error with http result %s", httpRes.Status)
+		return nil, err
+	}
+
+	keypair, ok := keyPairResponse.GetKeypairOk()
+	if !ok {
+		return nil, errors.New("Can not create keypair")
+	}
+	return keypair, nil
+}
+
+// GetKeypair retrieve keypair from keypairName
+func (s *Service) ReadKeyPair() (*osc.Keypair, error) {
+	readKeypairRequest := osc.ReadKeypairsRequest{
+		Filters: &osc.FiltersKeypair{
+			KeypairNames: &[]string{},
+		},
+	}
+	oscApiClient := s.scope.GetApi()
+	oscAuthClient := s.scope.GetAuth()
+	readKeypairResponse, httpRes, err := oscApiClient.KeypairApi.ReadKeypairs(oscAuthClient).ReadKeypairsRequest(readKeypairRequest).Execute()
+	if err != nil {
+		fmt.Printf("Error with http result %s", httpRes.Status)
+		return nil, err
+	}
+	keypaires, ok := readKeypairResponse.GetKeypairsOk()
+	if !ok {
+		return nil, errors.New("Error retrieving OAPIKeyPair")
+	}
+
+	if len(*keypaires) == 0 {
+		return nil, nil
+	} else {
+		keypair := *keypaires
+		return &keypair[0], nil
+	}
+}
+
+// DeleteKeyPair delete machine keypair
+func (s *Service) DeleteKeyPair(keyPairName string) error {
+	deleteKeypairRequest := osc.DeleteKeypairRequest{KeypairName: keyPairName}
```

Comment:
Put in same format as other ?
osc.DeleteKeypairRequest{
KeypairName: keyPairName
}